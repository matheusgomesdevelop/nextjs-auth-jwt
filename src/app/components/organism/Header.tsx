import React from 'react';

import useTheme from '@/core/hooks/useTheme';

import Avatar from '../atoms/Avatar';
import Icon from '../atoms/Icon';
import Link from '../atoms/Link';
import Button from '../atoms/Button';

const Header: React.FC = () => {
    const theme = useTheme();

    return (
        <header className="w-full container h-64 px-20 py-16 flex justify-between items-center">
            <Link href="https://github.com/matheusgrodrigues">
                <Avatar image="/images/a-avatar.jpeg" />
            </Link>
            <Button variant="transparent" onClick={theme.toggleTheme}>
                <Icon icon="pi-sun" />
            </Button>
        </header>
    );
};

export default Header;
