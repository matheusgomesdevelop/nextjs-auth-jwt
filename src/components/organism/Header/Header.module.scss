@import "../../../styles/util/breakpoints";

.header {
  width: 100%;
  max-width: 1280px;
  height: 72px;
  padding: var(--spacing-20) var(--spacing-16);
  @include centralizar;
  display: flex;
  justify-content: space-between;

  @include d(tablet) {
    height: 80px;
    padding: var(--spacing-24) var(--spacing-32);
  }
}
