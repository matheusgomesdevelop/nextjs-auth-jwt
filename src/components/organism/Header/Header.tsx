import React from 'react';

import styles from './Header.module.scss';

import useTheme from '@/core/hooks/useTheme';

import { Avatar, Button, Link, Icon } from '@/components/atoms';

const Header: React.FC = () => {
    const theme = useTheme();

    return (
        <header data-testid="header" className={styles.header} id="header">
            <Link href="https://github.com/matheusgrodrigues">
                <Avatar data-testid="avatar-testid" image="/images/a-avatar.jpeg" label={''} shape="circle" />
            </Link>
            <Button variant="transparent" onClick={theme.toggleTheme}>
                <Icon icon="pi-sun" />
            </Button>
        </header>
    );
};

export default Header;
