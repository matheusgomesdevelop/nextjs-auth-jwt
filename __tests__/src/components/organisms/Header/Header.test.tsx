import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import Header from '@/components/organism/Header/Header';

describe('Deve renderizar o Header corretamente', () => {
    it('Deve renderizar o Avatar, corretamente', () => {
        render(<Header />);

        const getAvatar = screen.getByTestId('avatar-testid');

        expect(getAvatar).toBeInTheDocument();
    });

    it('Deve preservar a estrutura visual do componente', () => {
        const three = renderer.create(<Header />).toJSON();

        expect(three).toMatchSnapshot();
    });
});
