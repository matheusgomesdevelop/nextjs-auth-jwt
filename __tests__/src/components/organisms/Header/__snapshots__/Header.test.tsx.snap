// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o Header corretamente Deve preservar a estrutura visual do componente 1`] = `
<header
  className="header"
  data-testid="header"
  id="header"
>
  <a
    className="link"
    href="https://github.com/matheusgrodrigues"
  >
    <div
      className="p-avatar p-component p-avatar-image p-avatar-circle"
      data-pc-name="avatar"
      data-pc-section="root"
      data-testid="avatar-testid"
      style={{}}
    >
      <img
        alt="avatar"
        data-pc-section="image"
        onError={[Function]}
        src="/images/a-avatar.jpeg"
      />
    </div>
  </a>
  <button
    className="button p-button p-component"
    data-pc-name="button"
    data-pc-section="root"
    data-testid="button-testid"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="icon pi pi-sun"
    />
  </button>
</header>
`;
